### PR TITLE
The actual number of subruns sometimes does not match with the one that appears in the RunSummary file. 

### PR DESCRIPTION
**Pull request by marialainez**
Get n_subruns by looking at the R0 files.

### CODE REVIEW

**Comment by morcuended** [comment]
This is worrisome. What is the difference you've found? Is it related to the number of streams? If so, I think it should be fixed in lstchain where we are just looking at the 1st stream https://github.com/cta-observatory/cta-lstchain/blob/0a4ceac95c5e362bc202912527553a865c34e96d/lstchain/scripts/lstchain_create_run_summary.py#L100

Do you propose to look at the filename and get the maximum number regardless of the stream?

**Comment by marialainez** [comment]
I think it is not necessarily related to the number of streams. For example, it happens for run 3514 of 20210121, where 2 subruns are missing and there is just one subrun for which there are less than 4 streams.

**Comment by morcuended** [comment]
Now I realize it might be related to the "discarded" last subruns. It sometimes happens that the last subrun is spoiled and cannot be processed. In these cases, I've manually changed the effective number of subruns that osa was supposed to analyze.

**Comment by marialainez** [comment]
Then I merge this PR?

**Review by morcuended — CHANGES_REQUESTED** [review]

File: osa/scripts/gain_selection.py [hunk 1]
```diff
@@ -63,22 +63,27 @@ def apply_gain_selection(date: str, output_basedir: Path = None):
...
         if ref_source=="UCTS" or ref_source=="TIB":
 
-            for subrun in range(n_subruns):
+            for subrun in range(n_subruns+1):
                 new_files = glob.glob(f"{r0_dir}/LST-1.?.Run{run_id:05d}.{subrun:04d}.fits.fz")
 
                 if len(new_files) != 4:
                     for file in new_files:
-                        output_file = output_dir / file.name
+                        output_file = output_dir / file
```
[the remark this answers is not in the document]
Comment (marialainez) [reply osa/scripts/gain_selection.py:86]: new_files is a list of strings

[the remark this answers is not in the document]
Comment (morcuended) [reply osa/scripts/gain_selection.py:86]: right, it is from glob module not Path glob

File: osa/scripts/gain_selection.py [hunk 2]
```diff
@@ -63,22 +63,27 @@ def apply_gain_selection(date: str, output_basedir: Path = None):
...
-        n_subruns = run["n_subruns"]
         ref_time = run["dragon_reference_time"]
         ref_counter = run["dragon_reference_counter"]
         module = run["dragon_reference_module_index"]
         ref_source = run["dragon_reference_source"].upper()
 
+        subrun_numbers = []
+        files = glob.glob(f"{r0_dir}/LST-1.?.Run{run_id:05d}.????.fits.fz")       
+        for file in files:
+            subrun_numbers.append(int(file[-12:-8]))
```
[the remark this answers is not in the document]
Comment (marialainez) [reply osa/scripts/gain_selection.py:74]: This difference is that for this glob the subrun number is not provided, but for the one below yes

**Pull request merged**